cmd/jobsink/main.go

```diff
@@ -104,8 +111,14 @@ func main() {
 
 	logger.Info("Starting the JobSink Ingress")
 
+	trustBundleConfigMapLister := configmapinformer.Get(ctx, eventingtls.TrustBundleLabelSelector).Lister().ConfigMaps(system.Namespace())
+	var h *Handler
+
 	featureStore := feature.NewStore(logging.FromContext(ctx).Named("feature-config-store"), func(name string, value interface{}) {
 		logger.Info("Updated", zap.String("name", name), zap.Any("value", value))
+		if flags, ok := value.(feature.Flags); ok && h != nil {
+			h.authVerifier = auth.NewVerifier(ctx, eventpolicyinformer.Get(ctx).Lister(), trustBundleConfigMapLister, flags)
```

Comment:
This is changing a variable which is shared across threads, we might want to use a mutex or atomic handles

Now that I think more, maybe adding this "update logic" to the verifier might encapsulate that and we won't duplicate "multi-threading" everywhere this is used

Reply:
@Cali0707 just as an idea, you could pass a configmapwatcher to `auth.NewVerifier()`, which then retriggers the `initOIDCProvider()` on changes

Reply:
I like that idea @creydr - let me try this tomorrow

Reply:
That came from @pierDipi :)